### PR TITLE
Fix expression normalize for user variables

### PR DESCRIPTION
**Pull request by GiniAppsPartners**
### Brief Summary of Changes
Fix normalize function to not trim user variables such as: $width to $w
Add tests to verify edge cases

#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [ ] Refactoring
- [ ] New feature
- [x] Bug fix
- [x] Adds more tests

#### Are tests included?
- [x] Yes
- [ ] No

#### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

**Review by adimiz1 — APPROVED** [review]

**Review by const-cloudinary — CHANGES_REQUESTED** [review]

File: Cloudinary/Classes/Core/Features/Helpers/CLDExpression.swift [hunk 1]
```diff
@@ -286,14 +291,19 @@ import Foundation
...
+            return String()
+        }
+
+        let key = removeExtraDashes(from: replaceAllExpressionKeys(in: currentKey))
+
         if currentValue.isEmpty {
             return "\(key)"
         }
-        let value = removeExtraDashes(from: replaceAllUnencodeChars(in: currentValue))
+        let value = removeExtraDashes(from: replaceAllUnicodeChars(in: currentValue))
```
[the remark this answers is not in the document]
Comment (GiniAppsPartners) [reply Cloudinary/Classes/Core/Features/Helpers/CLDExpression.swift]: Done

**Review by const-cloudinary — APPROVED** [review]
